### PR TITLE
Service Intentions L4 support

### PR DESCRIPTION
**Pull request by ashwin-venkatesh**
Add support for service intentions

- Skip enterprise tests as they don't currently have an image with service intentions

### CODE REVIEW

File: test/acceptance/tests/fixtures/crds/serviceintentions.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  name: svc1
+spec:
+  protocol: http
+---
+apiVersion: consul.hashicorp.com/v1alpha1
+kind: ServiceDefaults
+metadata:
+  name: svc2
+spec:
+  protocol: http
```
Comment (lkysow) [inline test/acceptance/tests/fixtures/crds/serviceintentions.yaml:13]: This shoudn't require http should it? Since it should also work for L4

**Review by lkysow — COMMENTED** [review]
👍 pending tests passing after go.mod is fixed

File: .circleci/config.yml [hunk 2]
```diff
@@ -141,6 +141,7 @@ jobs:
               -secondary-kubeconfig="$secondary_kubeconfig" \
               -debug-directory="$TEST_RESULTS/debug" \
               -consul-k8s-image=hashicorpdev/consul-k8s:crd-controller-base-latest \
+              -consul-image=hashicorpdev/consul
```
Comment (lkysow) [inline .circleci/config.yml:144]: You need to delete this I think.

**Review by lkysow — APPROVED** [review]

**Review by ishustava — APPROVED** [review]
Looks great!

**Pull request merged**